grid/src/Cabana_Grid_FastFourierTransform.hpp

```diff
@@ -568,8 +568,7 @@ class HeffteFastFourierTransform
             Kokkos::ViewAllocateWithoutInitializing( "fft_work" ),
             2 * fftsize );
         _workspace = Kokkos::View<Scalar* [2], memory_space>(
-            Kokkos::ViewAllocateWithoutInitializing( "workspace" ),
-            4 * fftsize );
+            "workspace", _fft->size_workspace() );
```

Comment:
Change makes sense - let me know if you're satisfied with the change. We'll want to add a comment along the lines of "Purposely using zero-init on allocation" to remind us not to make this mistake again

Reply:
I'm not sure I am. The initialization shouldn't be necessary. I suspect the problem was actually that fftsize is an int and 4*fftsize was overflowing it. Using _fft->size_workspace() avoids this overflow, and so I just coincidentally fixed the overflow. Let me test and submit a revised PR instead.

Reply:
Okay, I think the latest change to this PR fixes it. fftwork was stored as an int when fft sizes could overflow that. Also changed to use the heffte interface to determine the right workspace size to allocate.